query_optimizer/PhysicalGenerator.cpp

```diff
@@ -176,6 +177,8 @@ P::PhysicalPtr PhysicalGenerator::optimizePlan() {
     rules.emplace_back(new AttachLIPFilters());
   }
 
+  rules.push_back(std::make_unique<Partition>(optimizer_context_));
```

Comment:
In the single-node version on a `NUMA` machine, we needs this partition rule. Sure, I am adding a flag to switch this rule.

Regarding the first question, if I put the partition rule before `LIP` rules, how could I process `FilterJoin` with partitioned inputs?